init.py

```diff
@@ -112,11 +112,11 @@ def submodule_status(path, sha_commit):
       print('Initialize submodule ' + path + ' using git ... ')
       subprocess.run(['git', 'submodule', 'update', '--init', path], check = True, cwd = sys.path[0])
 
-      # Check that the SHA tag stored in this file matches the one stored in the git index
-      cur_sha_commit = status[1:].split(' ')[0]
-      if (cur_sha_commit != sha_commit):
-        print('SHA-1 tag stored in index does not match SHA tag stored in this script.')
-        sys.exit(1)
+    # Check that the SHA tag stored in this file matches the one stored in the git index
+    cur_sha_commit = status[1:].split(' ')[0]
+    if (cur_sha_commit != sha_commit):
+      print('SHA-1 tag stored in index does not match SHA tag stored in this script.')
+      sys.exit(1)
```

Comment:
Has the index been updated to have these new SHA tags for meson and ninja? I am still debugging #790 upon merging #753 #777 and this #798, and now when I try to build with meson, I get this error. Could be that my local copy still has some issues due to improper merge, or could be that this isn't updated yet

Reply:
Previously the tags in git and in the init.py script where not matching and the error was not thrown because the indentation was wrong ... try `git submodule update` or deleting `meson`, `ninja`, `codi`, `medi` in the `externals` folder.

Reply:
I continued to have this issue, so I simply reset the stored SHA in the index to the one that you have in init.py. Should be all good. Will be adjusted in #790

Reply:
It seems like you changed the version of ninja and meson in your branch. Maybe just make sure to use the newest ones if they work without any problem.